### PR TITLE
Modify loop for indicator.accumulate()

### PR DESCRIPTION
**Pull request by ejtovar**
Modify loop where indicator.accumulate() happens to compute correct values at the boundary. Had to modify a previous shallow water test. Also added a new test for this bug.

### CODE REVIEW

File: tests/shallow_water/verification-steady_incline-erk33-l9.prm [hunk 1]
```diff
@@ -0,0 +1,94 @@
+##
+#
+# Shallow water benchmark
+#
+# A 1D benchmark configuration consisting of a steady flow over an inclined 
+# plane with Manning's friction. See section 4.1 of [1] for details.
+#
+# This configuration uses an explicit ERK(3, 3, 1) timestepping. Expected
+# results are reported in shallow_water-steady_inclinde_1d-erk33-l9.baseline
```
Comment (tamiko) [inline tests/shallow_water/verification-steady_incline-erk33-l9.prm]:
```suggestion
# results are reported in shallow_water-steady_incline_1d-erk33-l9.baseline
```

File: source/hyperbolic_module.template.h [hunk 2]
```diff
@@ -348,6 +348,7 @@ namespace ryujin
 
             indicator.accumulate(js, U_j, c_ij);
 
+            /* Then skip diagonal. If col_idx == 0, we continue. */
```
Comment (tamiko) [inline source/hyperbolic_module.template.h]: You need
```
if (col_idx == 0)
  continue;
```
here in order to actually skip the diagonal.

File: source/hyperbolic_module.template.h [hunk 3]
```diff
@@ -337,9 +337,9 @@ namespace ryujin
 
           indicator.reset(i, U_i);
 
-          /* Skip diagonal. */
-          const unsigned int *js = sparsity_simd.columns(i) + stride_size;
-          for (unsigned int col_idx = 1; col_idx < row_length;
+          /* Don't skip diagonal for indicator */
```
Comment (tamiko) [inline source/hyperbolic_module.template.h]:
```suggestion
```

**Review by tamiko — CHANGES_REQUESTED** [review]

**Comment by tamiko** [comment]
@ejtovar Thanks for fixing!

**Comment by ejtovar** [comment]
@tamiko Made the appropriate changes.

**Review by tamiko — APPROVED** [review]

**Pull request merged**